### DISCUSSION TITLE
Keyball39 USB接続側のOLEDが反応しない

### DISCUSSION THREAD

**Discussion by gs1068** [Yowkees/keyball#353] [Q&A] [answered]
12Pピンソケットを基盤に半田付けした際にハンダを多く流し過ぎてしまい穴が詰まってしました。
そのため、ハンダ吸い取り線を使用して12Pピンソケットを新しいものに取り替えたのですが、OLEDが反応せずRemapも反応しません。
https://github.com/Yowkees/keyball/blob/main/keyball39/doc/rev1/buildguide_jp.md#3-6promicro%E3%81%AE%E3%81%AF%E3%82%93%E3%81%A0%E4%BB%98%E3%81%91

可能な限りハンダ吸い取り線を使用したため、力づくで引き抜いたりはしていません。
吸い取りを行っている間少しだけ長い時間こてを基盤に当て続けてしまった可能性はありますが、見た目でわかる程度の破損はしていなさどうです。

もし原因が分かりましたら、ご教示いただければ幸いです。
![IMG20230921204243](https://github.com/Yowkees/keyball/assets/74349651/dd3af5e6-204b-4d61-aa77-b22304ed1fcf)
![IMG20230921204304](https://github.com/Yowkees/keyball/assets/74349651/7950ab84-3c5a-4335-8433-d003a3d4cf0f)
![IMG20230921204314](https://github.com/Yowkees/keyball/assets/74349651/7a572d37-d96c-43c2-a0e4-7d4bf70d2280)
![IMG20230921201730](https://github.com/Yowkees/keyball/assets/74349651/bcb1bc36-03cc-424c-a539-bea915446e6a)
![IMG20230921204224](https://github.com/Yowkees/keyball/assets/74349651/733a20bc-c469-4f4f-a885-78b7732dcc01)

**Comment by gs1068** [comment]
その後12Pピンソケットにはんだを盛ったりOLEDつけ直しをしてみたのですが、特に変化なしでした。

**Comment by gs1068** [comment]
試しに手持ちのKeyball44の12Pコンスルーをはめてみたところ反応なしでした。
左側は12Pコンスルーをはめてみたところ正常に動作しました。
この場合は基盤破損で修復不可でしょうか？

**Comment by Yowkees** [comment, marked as answer]
ご連絡ありがとうございます。

光の加減かもしれませんが、下写真の丸で示した個所がショートしていないでしょうか。ProMicro裏の半田付けはスルーホールがジグザグに配置されており隣とショートしやすいので、この周辺の半田付けのご確認をお願いいたします。

![KB39_oledfix](https://github.com/Yowkees/keyball/assets/84209772/4b30fc9a-f5a2-4923-a0d5-5894c6be58fb)

また、REMAPはkeyball39_viaのファームウェアを書き込まないと動作しませんのでご確認ください。また、現在Keyballファームウェアを最新REMAPに対応中でして、当面は旧バージョンのREMAP（https://qmk018.remap-keys.app/）
を使用していただけますようお願いいたします。

以上、ご確認お願いいたします。

**Comment by gs1068** [comment]
ご確認ありがとうございます！
ハンダを全て除去してしまったので現物確認できませんでしたが、先ほど改めてハンダをきちんと除去して下記の条件確認してみたところOLEDが反応しました。
> 試しに手持ちのKeyball44の12Pコンスルーをはめてみたところ反応なしでした。

反応しなかった時にはマイコンが熱くなってしまっていたのでショートしていた可能性がありそうでした。
とても勉強になりました。ありがとうございます。

ただ12Pピンソケットを全てダメにしてしまったのでこの際12Pピンソケットに変更して組んでみようかと思います
